Requests push

ramon-rosado opened the pull request:
Finished Request Handler Implementation

ramon-rosado committed:
Started handler and routes for requests

ramon-rosado committed:
continued with handler
Get all request and get request by id work

ramon-rosado committed:
Finished implementing Request Handler

Steven6798 reviewed (approved):
Every method related to the requests where tested in my presence. No modifications where done to methods and files outside of our scope.

Ale573 committed:
Removed Heroku URL

Portela1 reviewed (approved):
I was present when the implementation occurred and reviewed all the changes / the methods were tested using postman and viewing the outcome directly on the database.

Portela1 committed:
Merge branch 'master' into Requests_PUSH

Everod52 commented on handler/request.py:
```diff
@@ -0,0 +1,112 @@
+                if request_to_update:
+                    valid_params = RequestHandler.verify_params(json, Request.REQUEST_REQUIRED_PARAMS)
+                    for key, value in valid_params.items():
+                        if key == 'supplyName':
+                            setattr(request_to_update, key, value)
+                        elif key == 'status':
+                            setattr(request_to_update, key, value)
+                        elif key == 'description':
+                            setattr(request_to_update, key, value)
```
Change to: (while you're at it, change it in the update_donation and update_user methods)
```suggestion
                        if key in Request.REQUEST_REQUIRED_PARAMS:
                            setattr(request_to_update, key, value)
```

Portela1 replied:
This is so you can't change time, uid and rid.
Both would work but we made that assumption that those params should not change.

Everod52 commented on handler/request.py:
```diff
@@ -0,0 +1,112 @@
+
+    @staticmethod
+    def get_request_by_id(rid):
+        if rid > 0:
```
Remove comparison with 0.

ramon-rosado replied:
Comparison with 0 is because there should never be a negative number rid. It will work but the front end group would have to be wary... and also when testing from postman (aka the professor)

Everod52 commented on handler/request.py:
```diff
@@ -0,0 +1,112 @@
+            return jsonify(message="Bad Request!"), 400
+
+    @staticmethod
+    def insert_request(json):
```
Change name to `create_request`

Everod52 commented on handler/request.py:
```diff
@@ -0,0 +1,112 @@
+
+
+class RequestHandler(BaseHandler):
+    # def __init__(self):
+    #     print("lol")
+
+    # def build_request_dict(self, row):  # row is a python list
+    #     result = {}
+    #     result['rid'] = row[0]
+    #     result['supplyName'] = row[1]
+    #     result['time'] = row[2]
+    #     result['status'] = row[3]
+    #     result['description'] = row[4]
+    #     result['uid'] = row[5]
+    #     result['did'] = row[6]
+    #     return result
```
Delete commented code

Everod52 commented on config/__init__.py:
```diff
@@ -4,7 +4,7 @@
 from flask_sqlalchemy import SQLAlchemy
 
-DEV_DB = 'postgresql://insouser:inso4115@localhost/DisasterAid'
+# DEV_DB = 'postgresql://insouser:inso4115@localhost/DisasterAid'
```
Uncomment this variable

Ale573 reviewed (approved):
Everything is working! Tested in Postman.

Everod52 commented on app.py:
```diff
@@ -74,5 +75,28 @@ def get_donation_by_id(did):
+    if request.method == 'GET':
+        return RequestHandler().get_all_requests()
+    elif request.method == 'POST':
+        return RequestHandler().insert_request(request.json)
```
Change method name here too

Everod52 commented on app.py:
```diff
@@ -74,5 +75,28 @@ def get_donation_by_id(did):
+        return jsonify(message="Method not allowed."), 405
+
+
+@app.route("/DAD/request/<int:rid>", methods=['GET', 'PUT', 'DELETE'])
```
Change url to `/DAD/requests/<int:rid>`

Everod52 commented:
When the changes are made, resolve the comments and then it can be merged.

Portela1 committed:
Update handler/request.py

Co-authored-by: Everson Rodriguez <34807511+Everod52@users.noreply.github.com>

ramon-rosado committed:
Updated DB config

Uncommented local DB and added a commented Heroku DB address for future reference

ramon-rosado committed:
Updated Request route and method name

Changed the route from "/DAD/request/<int:rid>" to "/DAD/requests/<int:rid>". Also changed the insert request method name to create_request()

Portela1 committed:
Changes requested

Portela1 committed:
Merge branch 'Requests_PUSH' of https://github.com/uprm-inso-4115-2019-2020-S2/semester-project-inso4115-DisasterAid-backend into Requests_PUSH

Everod52 commented on config/__init__.py:
```diff
@@ -5,6 +5,8 @@
 
 DEV_DB = 'postgresql://insouser:inso4115@localhost/DisasterAid'
+# LINE BELOW IS FOR WHEN WE WANT TO RUN ON HEROKU
+# DEV_DB = 'postgres://yukoojywyelupb:535a04c9549446512e582543c0d78e6844f1c2363eb646149845abf854592b65@ec2-3-230-106-126.compute-1.amazonaws.com:5432/dc08d0ste8gj9e'
```
Remove comment.

ramon-rosado committed:
Updated DB config

deleted heroku db config

ramon-rosado reviewed (commented):
submitted suggested changes

Everod52 reviewed (approved):
Well done!
